### DISCUSSION TITLE
Issue with deep query string parsing

### DISCUSSION THREAD

**Discussion by abdalem** [adonisjs/core#4293] [Help] [answered]
Hi everyone. I have an issue on one of my query. Some context, we are building a filter group feature and we want to allow, via a query string, to filter with a mongo like filter in query string. The first screen is the payload, the second one is the parsed query string. As you can see, at a given depth, the object is messed up. Thanks by advance guys !
![image](https://github.com/adonisjs/core/assets/19971392/b3e97094-6f23-44b3-bc02-082a9ce003cb)
![image](https://github.com/adonisjs/core/assets/19971392/44177038-eea6-4294-b132-50236643c296)

**Comment by aikrom** [comment, marked as answer]
Hey! AdonisJS uses the `qs` package to parse the query string and by default, when nesting objects, `qs` will only parse up to 5 children. In the controller, you can manually parse a query string of any depth.

```ts
import qs from 'qs'
import type { HttpContextContract } from '@ioc:Adonis/Core/HttpContext'

export default class TestController {
  public async index({ request }: HttpContextContract) {
    const query = qs.parse(request.parsedUrl.query, { depth: 10 });
  }
}
```

Also you can try to override the query string somewhere in the middleware (I haven't tested) with `Request.updateQs(...)`

[Here](https://github.com/adonisjs/http-server/blob/develop/src/Request/index.ts) you can find more about how query string parsing works.

**Reply by abdalem** [reply]
Thank you very much @aikrom

**Reply by aikrom** [reply]
It seems there was a way to set up query string parsing. This is covered in the [docs](https://docs.adonisjs.com/guides/request#configuring-query-string-parser) for v6, but will also work with v5.
